File: src/main/java/red/tetracube/smarthub/properties/SmartHubConfig.java
package red.tetracube.smarthub.properties;

import io.quarkus.arc.config.ConfigProperties;
import org.eclipse.microprofile.config.inject.ConfigProperty;

@ConfigProperties(prefix = "smart-hub")
public interface SmartHubConfig {

    @ConfigProperty(name = "iot")
    IoTConfig iot();
}

File: src/main/kotlin/iot/tetracubered/config/SmartHubConfig.kt
package iot.tetracubered.config

import io.quarkus.arc.config.ConfigProperties
import org.eclipse.microprofile.config.inject.ConfigProperty

@ConfigProperties(prefix = "smart-hub")
interface SmartHubConfig {

    @ConfigProperty(name = "messaging")
    fun messaging(): MessagingConfig
}